lib/amplitude_web.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:convert';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'web/amplitude_js.dart';
+import 'dart:async';
+
+class AmplitudeFlutterPlugin {
+  static void registerWith(Registrar registrar) {
+    final channel = MethodChannel(
+      'amplitude_flutter',
+      const StandardMethodCodec(),
+      registrar,
+    );
+    final pluginInstance = AmplitudeFlutterPlugin();
+    channel.setMethodCallHandler(pluginInstance.handleMethodCall);
+  }
+
+  /// Handles method calls over the MethodChannel of this plugin.
+  /// Note: Check the "federated" architecture for a new way of doing this:
+  /// https://flutter.dev/go/federated-plugins
+  Future<dynamic> handleMethodCall(MethodCall call) async {
+    var args = jsonDecode(call.arguments.toString());
+    var instanceName = args['instanceName'];
+    Amplitude amplitude = Amplitude.getInstance(instanceName);
+    switch (call.method) {
+      case "init":
+        {
+          String apiKey = args['apiKey'];
+          return amplitude.init(apiKey, args['userId'] ?? null);
+        }
+      case "logEvent":
+        {
+          String eventType = args['eventType'];
+          return amplitude.logEvent(eventType, args['eventProperties'] ?? null);
```

Comment:
Does current example covers event properties as well? If not we can cover it in next PR.

Reply:
Yes, it's in the example app. The dart Map needs to convert to JSObject before passing into the javascript function. I just made new changes to fix the eventProperties can not be passed into JS.

Reply:
<img width="1150" alt="Screen Shot 2021-11-09 at 3 54 58 PM" src="https://user-images.githubusercontent.com/13028329/141024393-81d15251-baee-4c0b-b9a7-060957ac06bb.png">